### ISSUE TITLE
[Web] Setting `autofocus` on checkbox doesn't have an effect in accessibility mode

### ISSUE THREAD

**Issue by nbayati** [flutter/flutter#118737]
Setting focus on the element based on the `hasFocus` flag has not been implemented for role managers such as `checkable` and `incrementable`.

Proposed solution: Create a new role called `focusable` and assign it to elements that have `isFocusable` set to true. In that role, set `tabIndex=0` if necessary, and call `.focus()` wrapped in a post update callback.

This to consider: 
- Don't assign this to `textfield` because it has its own logic for focus management.
- This class can potentially call `unfocus` if necessary.
- We can use this and implement a behavior similar to what currently google.com does where the input focus follows the a11y ring when the user is exploring the page with assistive technology. We can use [didGainAccessibilityFocus](https://api.flutter.dev/flutter/dart-ui/SemanticsAction/didGainAccessibilityFocus-constant.html) and [didLoseAccessibilityFocus](https://api.flutter.dev/flutter/dart-ui/SemanticsAction/didLoseAccessibilityFocus-constant.html).

<details>
<summary>Code sample</summary>

```
import 'package:flutter/material.dart';

void main() {
  runApp(MaterialApp(
    home: HomePage(),
  ));
}

class HomePage extends StatefulWidget {
  @override
  _HomePageState createState() => _HomePageState();
}

class _HomePageState extends State<HomePage> {
  bool value1 = false;
  bool value2 = false;
  bool value3 = false;

  @override
//App widget tree
  Widget build(BuildContext context) {
    return MaterialApp(
      home: Scaffold(
        body: Center(
          /** Card Widget **/
          child: Card(
            child: Padding(
              padding: const EdgeInsets.all(15.0),
              child: SizedBox(
                width: 430,
                height: 700,
                child: Column(
                  children: [
                    Row(
                      children: <Widget>[
                        Checkbox(
                          value: this.value1,
                          onChanged: (bool? value) {
                            setState(() {
                              this.value1 = value ?? false;
                            });
                          },
                        ),
                        Checkbox(
                          value: this.value2,
                          autofocus: true,
                          onChanged: (bool? value) {
                            setState(() {
                              this.value2 = value ?? false;
                            });
                          },
                        ),
                        Checkbox(
                          value: this.value3,
                          onChanged: (bool? value) {
                            setState(() {
                              this.value3 = value ?? false;
                            });
                          },
                        ),
                      ],
                    ),
                  ],
                ),
              ),
            ),
          ),
        ),
      ),
    );
  }
}
```
</details>
